app/src/main/java/com/codepath/apps/critterfinder/adapters/PetFavoritesAdapter.java

```diff
@@ -66,9 +67,8 @@ public ViewHolder(final View itemView) {
         @Override
         public void onClick(View v) {
             if (v instanceof ImageButton){
-                mOnItemClickListener.onButtonClicked((ImageButton)v,this,getAdapterPosition());
+                removePetAt(getAdapterPosition());
```

Comment:
What you had before was better. Before you were calling mOnItemClickListener.onItemClick and letting the activity update the data source and notify the adapter that the item has been removed. 

Generally speaking an adapter shouldn't have business logic in it like this. Curious why you  changed the implementation in this commit?